### DISCUSSION TITLE
ODataDynamicModel: Why {entityset}({key}) doesn't work?

### DISCUSSION THREAD

**Discussion by csharp4fun1** [OData/AspNetCoreOData#860] [Q&A] [answered]
Hello,

I tried the ODataDynamicModel because I have to switch from OData 7 to OData 8 and have dynamic data model coming at runtime. 

The example is working basically out of the box. So calling http://[localhost:4527/odata/mydatasource/products](http://localhost:4527/odata/mydatasource/products) is getting me the products where the products are dynamic data. 
https://github.com/OData/AspNetCoreOData/blob/0ab15a448285bb53e4d33a48f0feb6c163699d94/sample/ODataDynamicModel/Controllers/HandleAllController.cs#L30

But the odata endpoint below isn't working. When I try to call http://localhost:4527/odata/mydatasource/products(1) - that means this endpoint here
https://github.com/OData/AspNetCoreOData/blob/0ab15a448285bb53e4d33a48f0feb6c163699d94/sample/ODataDynamicModel/Controllers/HandleAllController.cs#L53
it doesn't go there. 

These examples are also written here on bottom of the page: https://github.com/OData/AspNetCoreOData/tree/main/sample/ODataDynamicModel

Does anyone know why the more detailed endpoints are not working?

**Comment by pluc77** [comment]
I can confirm the above mentioned behavior. MVC Routing is not working. I'm assuming it's safe to create an issue from this discussion.

**Comment by pluc77** [comment]
See Issue #884 for resolution.

**Comment by xuzhg** [comment, marked as answer]
I think it's answered in https://github.com/OData/AspNetCoreOData/issues/884#issuecomment-1499469080
